### PR TITLE
Tuple tag validation

### PR DESCRIPTION
**Pull request by farzad-sedghi**
https://github.com/spotify/scio/issues/4692

### CODE REVIEW

**Review by clairemcginty — COMMENTED** [review]
I think it would be simpler to do this in the SortedBucketIO.java class (both in the [`CoGbk` constructor + the `and` method](https://github.com/spotify/scio/blob/main/scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/SortedBucketIO.java#L138-L160). That way the changes are constrained to a single file, work for both SortedBucketSource + SortedBucketTransform, and will still be applied even if the user chooses to forego using our Scala SMB APIs. wdyt?

**Comment by farzad-sedghi** [comment]
> I think it would be simpler to do this in the SortedBucketIO.java class (both in the [`CoGbk` constructor + the `and` method](https://github.com/spotify/scio/blob/main/scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/SortedBucketIO.java#L138-L160). That way the changes are constrained to a single file, work for both SortedBucketSource + SortedBucketTransform, and will still be applied even if the user chooses to forego using our Scala SMB APIs. wdyt?

totally forgot this is our own IO layer :)

File: scio-smb/src/main/scala/com/spotify/scio/smb/util/SMBMultiJoin.scala [hunk 1]
```diff
@@ -32,20 +33,21 @@ final class SMBMultiJoin(private val self: ScioContext) {
...
   def sortMergeCoGroup[KEY: Coder, A: Coder, B: Coder, C: Coder, D: Coder, E: Coder](
-    keyClass: Class[KEY],
-    a: SortedBucketIO.Read[A],
-    b: SortedBucketIO.Read[B],
-    c: SortedBucketIO.Read[C],
-    d: SortedBucketIO.Read[D],
-    e: SortedBucketIO.Read[E],
-    targetParallelism: TargetParallelism
-  ): SCollection[(KEY, (Iterable[A], Iterable[B], Iterable[C], Iterable[D], Iterable[E]))] = {
+                                                                                      keyClass: Class[KEY],
```
Comment (shnapz) [inline scio-smb/src/main/scala/com/spotify/scio/smb/util/SMBMultiJoin.scala:36]: hmmmm.... why is there so much whitespace? is everything ok with styling?

**Comment by farzad-sedghi** [comment]
Closed because none of the changes in this PR were needed

**Pull request closed without merging**